pkg/expapi/types.go

```diff
@@ -319,3 +319,98 @@ type DaemonList struct {
 
 	Items []Daemon `json:"items"`
 }
+
+// Job represents the configuration of a single job.
+type Job struct {
+	api.TypeMeta   `json:",inline"`
+	api.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is a structure defining the expected behavior of a job.
+	Spec JobSpec `json:"spec,omitempty"`
+
+	// Status is a structure describing current status of a job.
+	Status JobStatus `json:"status,omitempty"`
+}
+
+// JobList is a collection of jobs.
+type JobList struct {
+	api.TypeMeta `json:",inline"`
+	api.ListMeta `json:"metadata,omitempty"`
+
+	Items []Job `json:"items"`
+}
+
+// JobSpec describes how the job execution will look like.
+type JobSpec struct {
+
+	// Parallelism specifies the maximum desired number of pods the job should
+	// run at any given time. The actual number of pods running in steady state will
+	// be less than this number when ((.spec.completions - .status.successful) < .spec.parallelism),
+	// i.e. when the work left to do is less than max parallelism.
+	Parallelism *int `json:"parallelism,omitempty"`
+
+	// Completions specifies the desired number of successfully finished pods the
+	// job should be run with. Defaults to 1.
+	Completions *int `json:"completions,omitempty" descrption:"desired number of pods"`
+
+	// Selector is a label query over pods that should match the pod count.
+	Selector map[string]string `json:"selector"`
+
+	// Template is the object that describes the pod that will be created when
+	// executing a job.
+	Template *api.PodTemplateSpec `json:"template"`
+}
+
+// JobStatus represents the current state of a Job.
+type JobStatus struct {
+
+	// Conditions represent the latest available observations of an object's current state
+	Conditions []JobCondition `json:"conditions,omitempty"`
+
+	// CreationTime represents time when the job was created
+	CreationTime util.Time `json:"creationTime,omitempty"`
+
+	// StartTime represents time when the job was started
+	StartTime util.Time `json:"startTime,omitempty"`
```

Comment:
should this be called StartTimestamp for consistency, and likewise below?
@bgrant0607